scripts/commands/setcraftrank.lua

```diff
@@ -0,0 +1,56 @@
+---------------------------------------------------------------------------------------------------
+-- func: setcraftRank <craft skill or ID> <craft rank> <target>
+-- desc: set specified craft skill RANK
+---------------------------------------------------------------------------------------------------
+require("scripts/globals/status")
+
+cmdprops =
+{
+    permission = 1,
+    parameters = "sss"
+}
+
+function error(player, msg)
+    player:PrintToPlayer(msg)
+    player:PrintToPlayer("!setcraftRank <craft skill or ID> <craft rank> {player}")
+end
+
+function onTrigger(player, craftName, tier, target)
+    if craftName == nil then
+        error(player, "You must specify a craft skill to set!")
+        return
+    end
+
+    local skillID = tonumber(craftName) or dsp.skill[string.upper(craftName)]
+    local targ = nil
+
+    if skillID < 48 or skillID > 57 then
+        error(player, "You must specify a valid craft skill.")
+        return
+    end
+
+    if target == nil then
+        targ = player
+    else
+        targ = GetPlayerByName(target)
+        if targ == nil then
+            player:PrintToPlayer(string.format("Player named '%s' not found!", target))
+            return
+        end
+    end
+
+    if tier == nil then
+        player:PrintToPlayer(string.format("%s's skillID %s rank: %u", targ:getName(), craftName, targ:getSkillRank(skillID)))
+        return
+    end
+
+    local craftRank = tonumber(tier) or dsp.craftRank[string.upper(tier)]
+    if craftRank == nil then
+        error(player, "Invalid craft rank!")
+        return
+    else
+        targ:setSkillRank(skillID, craftRank)
```

Comment:
maybe when targ:getName() == player:getName(), send the first message and not the second.

Reply:
good idea, will edit when I get free from work

Reply:
went to fix and realized I also forgot to handle nil skill/rank input while still having a player target. I could either move those to separate commands, so you'd have get and set..or do the argument shifting thing. I'll decide which and then come back to this.

Reply:
I didn't want to use a bunch of checking for "is this a craft or other skill?" and "did we omit param a but include param B anyway?" so I went with a  "just works" separate commands way of handling it. someone else can simplify/merge the commands later if they want.